### PR TITLE
feat:add color-picker component

### PR DESCRIPTION
**Pull request by cydiacen**
First of all, thank you for your contribution! 😄

New feature please send pull request to feature branch, and rest to master branch. Pull request will be merged after one of collaborators approve. Please makes sure that these form are filled before submitting your pull request, thank you!

[[中文版模板 / Chinese template](https://github.com/vueComponent/ant-design-vue/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### What's the background?

> 1. Describe the source of requirement.
> 2. Resolve what problem.
> 3. Related issue link.

### API Realization (Optional if not new feature)

> 1. Basic thought of solution and other optional proposal.
> 2. List final API realization and usage sample.
> 3. GIF or snapshot should be provided if includes UI/interactive modification.

### What's the effect? (Optional if not new feature)

> 1. Does this PR affect user? Which part will be affected?
> 2. What will say in changelog?
> 3. Does this PR contains potential break change or other risk?

### Changelog description (Optional if not new feature)

> 1. English description
> 2. Chinese description (optional)

### Self Check before Merge

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### Additional Plan? (Optional if not new feature)

> If this PR related with other PR or following info. You can type here.

### CODE REVIEW

File: components/color-picker/ColorPicker.jsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+      };
+      result.lang = {
+        ...result.lang,
+        ...(this.$props.locale || {}).lang,
+      };
+      return result;
+    },
+    renderColorPicker(){
+      const { prefixCls: customizePrefixCls } = this.$props;
+      const getPrefixCls = this.configProvider.getPrefixCls;
```
Comment (Amour1688) [inline components/color-picker/ColorPicker.jsx:90]: 解构

File: components/color-picker/locale/zh_CN.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export default {
+  save: 'Save',
+  clear: 'Clear',
+  cancel: 'Cancel',
```
Comment (Amour1688) [inline components/color-picker/locale/zh_CN.js:4]: 中文是不是要改成保存

File: components/color-picker/ColorPicker.jsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+export default {
+  name: 'AColorPicker',
+  mixins: [BaseMixin],
+  model: {
+    prop: 'value',
+    event: 'change',
+  },
+  props: {
+    prefixCls: PropTypes.string,
+    config:PropTypes.object,//pickr配置
```
Comment (Amour1688) [inline components/color-picker/ColorPicker.jsx:27]: 代码需要跑一下 lint

File: components/color-picker/style/index.less [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+      }
+    }
+    .@{color-picker-prefix-cls}-selection {
+      // .active();
+    }
+  }
+  &-selection{
+    display: block;
+    box-sizing: border-box;
+    background-color: #fff;
```
Comment (Amour1688) [inline components/color-picker/style/index.less:43]: 可以用全局的变量

File: components/color-picker/locale/zh_TW.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+export default {
+  save: 'Save',
+  clear: 'Clear',
+  cancel: 'Cancel',
```
Comment (Amour1688) [inline components/color-picker/locale/zh_TW.js:4]: 这边也要改成中文吧

Comment (cydiacen) [reply components/color-picker/locale/zh_TW.js:4]: 语言包的功能,pickr的适配还有些问题,所以就没改全了.等pickr那边的issue有回复解决方案,再一并都改改好

File: components/color-picker/ColorPicker.jsx [hunk 6]
```diff
@@ -0,0 +1,129 @@
+import PropTypes from '../_util/vue-types';
+import { ConfigConsumerProps } from '../config-provider';
+import BaseMixin from '../_util/BaseMixin';
+import '@simonwep/pickr/dist/themes/classic.min.css';   // 'classic' theme
```
Comment (tangjinzhou) [inline components/color-picker/ColorPicker.jsx:4]: 放在 less 文件里面吧

File: components/color-picker/ColorPicker.jsx [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+      myOpen: false,
+      pickr: null,
+    };
+  },
+  mounted() {
+    this.createPickr();
+  },
+  methods: {
+    createPickr() {
+      this.pickr = Pickr.create(Object.assign({
```
Comment (tangjinzhou) [inline components/color-picker/ColorPicker.jsx:46]: 可以支持所有事件 https://github.com/Simonwep/pickr#events

File: components/color-picker/ColorPicker.jsx [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+        el: '#color-picker' + this._uid,
```
Comment (tangjinzhou) [inline components/color-picker/ColorPicker.jsx:47]: https://github.com/Simonwep/pickr#options
// Where the pickr-app should be added as child.
    container: 'body',
container 支持组件库的api，参考select  getPopupContainer

File: components/color-picker/ColorPicker.jsx [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+      };
+      if (sizeOpt[this.size]) {
+        classString[sizeOpt[this.size]] = true;
+      }
+      return (
+        <div
+          class={classString}
+          tabIndex={props.disabled ? -1 : 0}
+          onClick={this.handleOpenChange.bind(this, !this.myOpen)}
+          {...props}
```
Comment (tangjinzhou) [inline components/color-picker/ColorPicker.jsx:110]: {...props} 这里没必要加了吧

File: components/color-picker/ColorPicker.jsx [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+  name: 'AColorPicker',
+  mixins: [BaseMixin],
+  model: {
+    prop: 'value',
+    event: 'change',
+  },
+  props: {
+    prefixCls: PropTypes.string,
+    config: PropTypes.object,//pickr配置
+    value: PropTypes.string,//颜色值
```
Comment (tangjinzhou) [inline components/color-picker/ColorPicker.jsx:26]: 新增 defaultValue，相关概念搜索 受控组件/非受控组件

File: components/color-picker/ColorPicker.jsx [hunk 11]
```diff
@@ -0,0 +1,129 @@
...
+        'small': `${prefixCls}-sm`,
+      };
+      if (sizeOpt[this.size]) {
+        classString[sizeOpt[this.size]] = true;
+      }
+      return (
+        <div
+          class={classString}
+          tabIndex={props.disabled ? -1 : 0}
+          onClick={this.handleOpenChange.bind(this, !this.myOpen)}
```
Comment (Amour1688) [inline components/color-picker/ColorPicker.jsx:109]: 为啥不直接在 `handleOpenChange` 里取 `this.myOpen`

File: components/color-picker/ColorPicker.jsx [hunk 12]
```diff
@@ -18,15 +15,17 @@ export default {
   mixins: [BaseMixin],
   model: {
     prop: 'value',
-    event: 'change',
+    event: 'myChange',//为了支持v-model直接返回颜色字符串 所以用了自定义的事件,与pickr自带change事件进行区分
```
Comment (tangjinzhou) [inline components/color-picker/ColorPicker.jsx:18]: 名称改成 change.value 吧

File: components/color-picker/ColorPicker.jsx [hunk 13]
```diff
@@ -40,13 +39,23 @@ export default {
   },
   mounted() {
     this.createPickr();
+    this.eventsBinding();
   },
   methods: {
+    eventsBinding() {
+      Object.keys(this.$listeners).forEach(event => {
```
Comment (tangjinzhou) [inline components/color-picker/ColorPicker.jsx:46]: 不是所有的事件都绑定到 picker 吧  如果绑定了picker本身不支持的会不会有问题   例如 click dbclick ...

File: components/color-picker/ColorPicker.jsx [hunk 14]
```diff
@@ -106,8 +116,8 @@ export default {
         <div
           class={classString}
           tabIndex={props.disabled ? -1 : 0}
-          onClick={this.handleOpenChange.bind(this, !this.myOpen)}
-          {...props}
+          onClick={this.handleOpenChange}
+          {...this.$attrs}
```
Comment (tangjinzhou) [inline components/color-picker/ColorPicker.jsx:120]: 这个也没必要加   vue会自动将没有声明的属性，挂载到根结点上 ，除非明确声明了 inheritAttrs: false

File: components/color-picker/locale/zh_TW.js [hunk 15]
```diff
@@ -1,5 +1,5 @@
 export default {
-  save: 'Save',
-  clear: 'Clear',
-  cancel: 'Cancel',
+  save: '保存',
```
Comment (tangjinzhou) [inline components/color-picker/locale/zh_TW.js:2]: 繁体字 ？

File: components/color-picker/style/index.less [hunk 16]
```diff
@@ -1,6 +1,8 @@
 @import '../../style/themes/index';
 @import '../../input/style/mixin';
-
+@import '../../../node_modules/@simonwep/pickr/dist/themes/classic.min.css';   // 'classic' theme
```
Comment (tangjinzhou) [inline components/color-picker/style/index.less:3]: 为啥用相对路径，
这种支持的吧 @import '～@simonwep/pickr/dist/themes/classic.min.css';

File: components/color-picker/ColorPicker.jsx [hunk 17]
```diff
@@ -18,15 +15,17 @@ export default {
...
   model: {
     prop: 'value',
-    event: 'change',
+    event: 'myChange',//为了支持v-model直接返回颜色字符串 所以用了自定义的事件,与pickr自带change事件进行区分
   },
   props: {
     prefixCls: PropTypes.string,
+    defaultValue:PropTypes.string,//默认值
     config: PropTypes.object,//pickr配置
     value: PropTypes.string,//颜色值
```
Comment (tangjinzhou) [inline components/color-picker/ColorPicker.jsx:24]: 1、外层value 变化，没有响应  
watch: {
 value(val) {
 picker.setColor
}
}
2、还有 config  的数据变化后如何同步？
3、新增 diable、color-format（参考element） 的属性

File: components/color-picker/ColorPicker.jsx [hunk 18]
```diff
@@ -40,13 +39,23 @@ export default {
   },
   mounted() {
```
Comment (tangjinzhou) [inline components/color-picker/ColorPicker.jsx:40]: 组件销毁后 picker 实例要同步销毁

File: components/color-picker/ColorPicker.jsx (hunk 13 above)
Comment (cydiacen) [reply components/color-picker/ColorPicker.jsx:46]: 好的 那我维护个事件数组

File: components/color-picker/ColorPicker.jsx [hunk 19]
```diff
@@ -0,0 +1,139 @@
...
+      const props = getOptionProps(this);
+      const classString = {
+        [`${prefixCls}-box`]: true,
+        [`${prefixCls}-open`]: this.myOpen,
+      };
+      const sizeOpt = {
+        'large': `${prefixCls}-lg`,
+        'small': `${prefixCls}-sm`,
+      };
+      if (sizeOpt[this.size]) {
```
Comment (Amour1688) [inline components/color-picker/ColorPicker.jsx:112]: 这部分的逻辑是不是可以直接写到 `classString` 的声明中

File: components/color-picker/locale/zh_TW.js (hunk 15 above)
Comment (cydiacen) [reply components/color-picker/locale/zh_TW.js:2]: 我查了下,这几个的繁体 也是长这样的

**Pull request closed without merging**